### DISCUSSION TITLE
Pnpm install not working using AWS CodeArtifact in docker container

### DISCUSSION THREAD

**Discussion by cqConvene** [pnpm/pnpm#6305] [Q&A] [answered]
Hello team,  I have a curious question and I am not entirely sure its package manager related. 

I am trying to create a docker image and we run the pnpm install in a builder.  One of the steps before is to pull down the AWS credentials.  It creates the .npmrc file and sets the auth token.  However the installs will fail because of authorization.  

![Screenshot 2023-03-28 at 5 10 24 PM](https://user-images.githubusercontent.com/62907600/228367948-cba2efdb-c504-4f08-b4ae-1f750cd490a7.png)

![Screenshot 2023-03-28 at 5 12 52 PM](https://user-images.githubusercontent.com/62907600/228368518-ec073f09-df46-4fa7-8bfd-3d7739b967e8.png)

When we switch it to use NPM instead it was able to install.   What am I doing incorrectly?

**Comment by gluxon** [comment, marked as answer]
My initial suggestion is to inspect the generated `.npmrc` and see if the `registry` value matches the `_authToken` config.

Pulling an example from the [Amazon CodeArtifact docs](https://docs.aws.amazon.com/codeartifact/latest/ug/npm-auth.html).

```npmrc
# .npmrc
registry=https://my_domain-111122223333.d.codeartifact.region.amazonaws.com/npm/my_repo/
//my_domain-111122223333.d.codeartifact.region.amazonaws.com/npm/my_repo/:_authToken=$CODEARTIFACT_AUTH_TOKEN
```

---

Taking a blind guess, but I'm noticing the request `GET` request URL has `:443` at the end while the authorization setting URL does not. Is there an extra `:443` at the end of the `registry` config? Does removing it still cause this to happen?

If removing the `:443` fixes this, there might be a pnpm bug matching `*.d.codeartifact.us-east-1.amazonaws.com:443` to `//d.codeartifact.us-east-1.amazonaws.com/npm/.../:_authToken=...`.

**Reply by cqConvene** [reply]
my .npmrc file with the registry settings does not have the :443

**Reply by cqConvene** [reply]
that was it, we added the :443 to our registry url in the .npmrc and it worked
